crates/task-impls/src/consensus/view_change.rs

```diff
@@ -91,7 +86,15 @@ pub(crate) async fn update_view<TYPES: NodeType, I: NodeImplementation<TYPES>>(
             )
             .await;
         }
-    }));
+    });
+
+    // Cancel the old timeout task
+    cancel_task(std::mem::replace(
```

Comment:
Does this cancel by swapping the memory and GCing the dangling reference? I'm not sure how each executor handles killing tasks, I assume it's fine though.

Reply:
The `std::mem::replace` atomically replaces the first (dest) argument with the second (src) and returns the first (old dest)

Since the first (dest) contained the old timeout task, we can now run `cancel_task` on it

Reply:
> Does this cancel by swapping the memory and GCing the dangling reference? I'm not sure how each executor handles killing tasks, I assume it's fine though.

It's worth noting there's definitely something to worry about in general: dropping a `JoinHandle` means the task will continue running until it shuts itself down, and we can no longer cancel it. This is basically never what we want.

But this looks right to me.

Reply:
> Does this cancel by swapping the memory and GCing the dangling reference? I'm not sure how each executor handles killing tasks, I assume it's fine though.

It's worth noting there's definitely something to worry about in general: dropping a `JoinHandle` means the task will continue running until it shuts itself down, and we can no longer cancel it. This is basically never what we want.

But this looks right to me.